common/main/cpp/native_c4index.cc

```diff
@@ -0,0 +1,195 @@
+        JNIEnv *env,
+        jclass ignore,
+        jlong handle) {
+#ifdef COUCHBASE_ENTERPRISE
+    auto idx = (C4Index *) handle;
+    if (idx != nullptr)
+        c4index_release(idx);
+#endif
+}
+
+/*
+ * Class:     com_couchbase_lite_internal_core_impl_NativeC4IndexUpdater
+ * Method:    count
+ * Signature: (J)J
+ */
+JNIEXPORT jlong JNICALL
+Java_com_couchbase_lite_internal_core_impl_NativeC4IndexUpdater_count(
+        JNIEnv *env,
+        jclass ignore,
+        jlong handle) {
+#ifndef COUCHBASE_ENTERPRISE
+    return 0;
+#else
+    return c4indexupdater_count((C4IndexUpdater *) handle);
+#endif
+}
+
+/*
+ * Class:     com_couchbase_lite_internal_core_impl_NativeC4IndexUpdater
+ * Method:    valueAt
+ * Signature: (JJ)J
+ */
+JNIEXPORT jlong JNICALL
+Java_com_couchbase_lite_internal_core_impl_NativeC4IndexUpdater_valueAt(
+        JNIEnv *env,
+        jclass ignore,
+        jlong handle,
+        jlong index) {
+#ifndef COUCHBASE_ENTERPRISE
+    return 0;
+#else
+    return (jlong) c4indexupdater_valueAt((C4IndexUpdater *) handle, (size_t) index);
+#endif
+}
+
+/*
+ * Class:     com_couchbase_lite_internal_core_impl_NativeC4IndexUpdater
+ * Method:    setVectorAt
+ * Signature: (JJ[F)V
+ */
+JNIEXPORT void JNICALL
+Java_com_couchbase_lite_internal_core_impl_NativeC4IndexUpdater_setVectorAt(
+        JNIEnv *env,
+        jclass ignore,
+        jlong handle,
+        jlong index,
+        jfloatArray jvalues) {
+#ifdef COUCHBASE_ENTERPRISE
+    jsize len = 0;
+    jfloat *val = nullptr;
+
+    if (jvalues != nullptr) {
+        len = env->GetArrayLength(jvalues);
+        val = env->GetFloatArrayElements(jvalues, nullptr);
+    }
+
+    C4Error error{};
+    bool ok = c4indexupdater_setVectorAt((C4IndexUpdater *) handle, (size_t) index, val, len, &error);
+
+    if (val != nullptr)
+        env->ReleaseFloatArrayElements(jvalues, val, 0);
+
+    if (!ok && error.code != 0)
+        throwError(env, error);
+#endif
+}
+
+/*
+ * Class:     com_couchbase_lite_internal_core_impl_NativeC4IndexUpdater
+ * Method:    skipVectorAt
+ * Signature: (JJ)Z
+ */
+JNIEXPORT jboolean JNICALL
+Java_com_couchbase_lite_internal_core_impl_NativeC4IndexUpdater_skipVectorAt(
+        JNIEnv *env,
+        jclass ignore,
+        jlong handle,
+        jlong index) {
+#ifndef COUCHBASE_ENTERPRISE
+    return JNI_FALSE;
+#else
+    return (c4indexupdater_skipVectorAt((C4IndexUpdater *) handle, (size_t) index) ? JNI_TRUE : JNI_FALSE);
+#endif
+}
+
+/*
+ * Class:     com_couchbase_lite_internal_core_impl_NativeC4IndexUpdater
+ * Method:    finish
+ * Signature: (J)V
+ */
+JNIEXPORT void JNICALL
+Java_com_couchbase_lite_internal_core_impl_NativeC4IndexUpdater_finish(
+        JNIEnv *env,
+        jclass ignore,
+        jlong handle) {
+#ifdef COUCHBASE_ENTERPRISE
+    C4Error error{};
+    bool ok = c4indexupdater_finish((C4IndexUpdater *) handle, &error);
+    if (!ok && error.code != 0) {
+        throwError(env, error);
+        return;
```

Comment:
I saw a comment in Slack indicating that you should release in an error situation as well.  It looks like you are not doing that here?

Reply:
Yeah... It turns out that test #22 explicitly forbids doing that.  Guess how I found out....